app/controllers/application_controller.rb

```diff
@@ -2,9 +2,140 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+	configure do
+		set :public_folder, 'public'
+		set :views, 'app/views'
+		enable :sessions
+		set :session_secret, "password_security"
+	end
+
+	get '/' do 
+		erb :index
+	end
+
+	get '/tweets' do 
+		if logged_in?
+			erb :'tweets/tweets'
+		else
+			redirect '/login'
+		end
+	end
+
+	get '/signup' do 
+		if self.logged_in?
+			redirect '/tweets'
+		else
+			erb :'users/create_user'
+		end
+	end
+
+	get '/login' do
+		if logged_in?
+			redirect '/tweets'
+		else
+			erb :'users/login'
+		end
+	end
+
+	get '/logout' do 
+		if logged_in?
+			session.clear
+			redirect '/login'
+		else
+			redirect '/'
+		end
+	end
+
+	get '/users/:slug' do 
+		@user = User.find_by_slug(params[:slug])
+		erb :'users/user_tweets'
+	end
+
+
+	post '/signup' do 
+		@user = User.new
```

Comment:
Except for the instance variable part.  Although it is not needed in a view currently, that could change in future versions, and simply making everything an instance variable in the beginning could save some time and confusion later.  Is this approach possibly a security concern?